src/chapters/classes-part2/instance-and-static-methods.rst

```diff
@@ -0,0 +1,235 @@
+Instance and Static Methods
+============================
+
+We explored configuring *data* within classes with fields and properties. Now
+let’s turn our attention to *behavior* (methods).
+
+.. index:: method
+
+Calling Methods on Objects
+--------------------------
+
+A *method* is a procedure that belongs to a class. In Java, all procedures must
+be part of a class---they cannot stand on their own. Let’s revisit our minimal
+class example.
+
+.. sourcecode:: java
+   :linenos:
+
+   public class HelloWorld {
+
+      private String message = "Hello World";
+
+      void sayHello() {
+         System.out.println(message);
+      }
+
+   }
+
+There is one method in this class, ``sayHello``. In order to call this method,
+we must create an object from the ``HelloWorld`` class template. In other
+words, we must have an instance of ``HelloWorld``.
+
+Here’s how you call methods on an object.
+
+.. sourcecode:: java
+   :linenos:
+
+   HelloWorld hello = new HelloWorld();
+   hello.sayHello();
+
+It is not possible to call ``sayHello`` without having a ``HelloWorld``
+object. This begins to make more sense when you note that the
+``message`` field is used within ``sayHello``, and unless we are calling
+``sayHello`` on an instantiated object, there will be no ``message``
+field available to print.
+
+.. admonition:: Tip
+
+   As mentioned before, class members should have the most restrictive level
+   of access possible. This goes for methods as well as fields.
+
+   For example, if you create a method that should *only* be used within
+   your class, then you should declare it ``private``. Think of ``private``
+   methods as those that are NOT useful outside of the class but contribute
+   internally to helping the class behave as desired or expected.
+
+   ``public`` methods contain code that other classes need to use when they
+   implement the class containing those methods. Make methods ``public`` only
+   when you expect other classes to use them, and when you are committed to
+   maintaining those methods for other programs that might need them.
+
+.. index:: ! instance method
+
+Instance Methods
```

Comment:
Yep.  I have no problem with putting this section in lesson 3 vs. lesson 4, but I think we should keep instance methods and static methods together.  The topics feed off of each other.

One concern I do have with putting IMs in lesson 3 is that the section hints at work to be done in the exercises, "The framework for these new methods is shown in the Student class below, but each method is missing some code. Filling in that code is left for you to do as one of the chapter exercises."

Since IMs were originally in part 2 of classes in the old curriculum, the exercises in our new lesson 3 do not address this. However, the exercises in lesson 4 do.

As I said, I have no strong preference for where to put this section, but we will have to make sure we update the class 3 and 4 exercises to match our choice.

Reply:
We can talk about this on Tuesday!